### ISSUE TITLE
[Bug] unexpected behavior when `ABBR_EXPANSION_CURSOR_MARKER` set to caret

### ISSUE THREAD

**Issue by olets** [olets/zsh-abbr#140]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Update the issue title

- [X] I have updated the title

### Expected Behavior

Can use an arbitrary string as the `ABBR_EXPANSION_CURSOR_MARKER`.

### Actual Behavior

If `ABBR_EXPANSION_CURSOR_MARKER` is `^`, `\^`, `'^'`, expansions without the cursor marker are doubled, with the cursor between them.

### Steps To Reproduce

```
% ABBR_EXPANSION_CURSOR_MARKER='^'
% abbr a=b
% a[SPACE] # expands to `b[CURSOR]b`
```

### Environment

```shell
zsh-abbr version 5.8.0
zsh 5.9 (arm-apple-darwin21.3.0)
OSTYPE darwin21.3.0
```

### Installation method

Plugin manager

### Installation method details

zcomet

### Anything else?

- Originally reported in https://github.com/olets/zsh-abbr/issues/84?show_full=true#issuecomment-2170611531

**Comment by olets** [comment]
> maybe the documentation could just suggest a couple of characters that are known to work without issues

Good idea. Rather than doing that exactly, I've added callouts for a few limitations

- https://github.com/olets/zsh-abbr-v5-docs/commit/619e4b69491839d262b103e8f2c143683552d088
- https://zsh-abbr.olets.dev/commands.html#add
- https://zsh-abbr.olets.dev/commands.html#erase
- https://zsh-abbr.olets.dev/usage.html#place-the-cursor
- https://zsh-abbr.olets.dev/advanced.html#configuration-variables

**Comment by olets** [comment]
The upcoming v6.0.0 will fix this

**Comment by olets** [comment]
Cool yeah I like the idea of using a Unicode character that I'd have to paste, that I can't type

Released in v6.0.0
